crates/bevy_ecs/src/system/commands/fallible.rs

```diff
@@ -0,0 +1,100 @@
+use super::Command;
+use crate::world::World;
+use bevy_utils::tracing::error;
+use std::fmt::Debug;
+
+pub struct CommandError<'a, C: FallibleCommand> {
+    pub error: C::Error,
+    pub world: &'a mut World,
+}
```

Comment:
@mockersf 
there's a chance we could incorporate the system name into this structure which could potentially replace #2222

Reply:
I think that's a great idea if possible.

Reply:
Is this something we want to hide behind a feature flag, or just something we should expose always.

Reply:
I don't see any reason not to expose it always? You can completely disable error logging already using .ignore right? That should contain the perf impact for when you're spawning a billion entities and need to squeeze out every drop of perf.

Reply:
always please! I put mine behind a feature flag because of the ugly panic catching

Reply:
Just a note, in order to do this, we'll have to alter the current `Command` trait, which is a breaking change, (if we're ok with this, then I am as well)

Reply:
@mockersf @alice-i-cecile 
To support this we need to rework the `Command` trait to be 
```rust
pub struct CommandArgs<'a> {
    world: &'a mut World,
    system_name: &'a Cow<'static, str>,
}

pub trait Command {
    fn write(self: Box<self>, args: CommandArgs);
}
```

This is probably best as its own PR however. 
It this a change that we would potentially desire?